plugin/plugin.go

```diff
@@ -23,6 +26,26 @@ func New(policy string) validator.Plugin {
 
 func (p *plugin) Validate(ctx context.Context, req *validator.Request) error {
 
+	log.SetFormatter(&log.JSONFormatter{})
+	log.SetOutput(os.Stdout)
+
+	promotion := Promotion{
+		Build:       req.Build.Parent,
+		Environment: req.Build.Deploy,
+	}
+
+	// log each request for audit purposes
+	log.WithFields(log.Fields{
+		"type":       "audit",
+		"extension":  "validation",
+		"user":       req.Build.Sender,
+		"event":      req.Build.Event,
+		"repository": req.Repo.Slug,
+		"branch":     req.Repo.Branch,
+		"commit":     req.Build.After,
+		"promotion":  &promotion,
+	}).Info(fmt.Sprintf("audit(%d)", time.Now().Unix()))
```

Comment:
It's a little unclear what drove the decision to use these specific fields, and manually managing these mappings seem like they could get out of sync with the underlying request fields in potentially annoying ways. 

I can understand ignoring the pipeline `Config.Data`, since that would be a fairly large log event, but why not just log everything from the underlying [Build and Repo types](https://github.com/drone/drone-go/blob/master/drone/types.go)?

Reply:
I tried to pick the main fields that would allow us to be compliant with ISO-27001 in a way that we can rebuild an entire trail should we need to.

Build and Repo types are pretty large in terms of attributes and I'm not sure that it would help engineers to analyse entries efficiently without previous knowledge of the Drone API. Some attributes are a bit overlapping (like repo name, namespace and slug) and others not relevant for an audit trail like `AuthorAvatar`, `Title` or `Status` that will always be pending at this stage of the pipeline.

We can surely include more attributes to the list but I don't feel that we should be dumping everything that comes in `req.Repo` and `req.Build` without filtering what kind of fields are relevant or not to our use case (SOX compliance).

Another option is to make it configurable in runtime and avoid an opinionated set, but I'm not sure that the benefits worth the extra complexity.